compute/cloud-client/src/main/java/compute/CreateInstance.java

```diff
@@ -88,21 +89,21 @@ public static void createInstance(String project, String zone, String instanceNa
       // Insert the instance in the specified project and zone.
       Operation response = instancesClient.insert(project, zone, instanceResource);
 
-      if (response.getStatus() == Status.RUNNING) {
-        // Wait for the create operation to complete; default timeout is 2 mins
-        response = zoneOperationsClient.wait(project, zone, String.valueOf(response.getId()));
+      // Wait for the create operation to complete.
+      // Timeout is set at 180000ms or 3 minutes.
+      long startTime = System.currentTimeMillis();
+      while (response.getStatus() != Status.DONE
+          && System.currentTimeMillis() - startTime < 180000) {
+        // The default wait timeout is 2 mins.
+        response = zoneOperationsClient.get(project, zone, String.valueOf(response.getId()));
```

Comment:
This is using `.get` instead of `.wait`. [Should we be using wait instead](https://cloud.google.com/compute/docs/reference/rest/v1/zoneOperations/wait)? 

If we are using wait and it has a 2 minute timeout, we can probably just try a max of 2-3 time or something and get the same effect.

Reply:
I don't have a strong preference here, but it does seem like `wait` would be less quota usage?

Reply:
'wait' method in Java leads to SocketTimeout exception. This issue will be fixed and released as part of LRO.
In the meantime, to demo the concept, 'get' is being used here.